iscsi/get.go

```diff
@@ -0,0 +1,283 @@
+			iopsPath, err)
+	}
+
+	return readmb, writemb, iops, nil
+}
+
+// GetFileioUdev is getting the actual info to build up
+// the FILEIO data and match with the enable target
+func (fileio FILEIO) GetFileioUdev(targetCorePath string, fileioNumber string,
+	objectName string) (fio *FILEIO, err error) {
+
+	fileio.Name = "fileio_" + fileioNumber
+	fileio.Fnumber = fileioNumber
+	fileio.ObjectName = objectName
+
+	udevPath := filepath.Join(targetCorePath, fileio.Name, fileio.ObjectName, "udev_path")
+
+	if _, err := os.Stat(udevPath); os.IsNotExist(err) {
+		return nil, fmt.Errorf("iscsi: GetFileioUdev: fileio_%s is missing file name", fileio.Fnumber)
+	}
+	filename, err := ioutil.ReadFile(udevPath)
+	if err != nil {
+		return nil, fmt.Errorf("iscsi: GetFileioUdev: Cannot read filename from udev link :%s", udevPath)
+	}
+	fileio.Filename = strings.TrimSpace(string(filename))
+
+	return &fileio, nil
+}
+
+// GetIblockUdev is getting the actual info to build up
+// the IBLOCK data and match with the enable target
+func (iblock IBLOCK) GetIblockUdev(targetCorePath string, iblockNumber string,
+	objectName string) (ib *IBLOCK, err error) {
+
+	iblock.Name = "iblock_" + iblockNumber
+	iblock.Bnumber = iblockNumber
+	iblock.ObjectName = objectName
+
+	udevPath := filepath.Join(targetCorePath, iblock.Name, iblock.ObjectName, "udev_path")
+
+	if _, err := os.Stat(udevPath); os.IsNotExist(err) {
+		return nil, fmt.Errorf("iscsi: GetIBlockUdev: iblock_%s is missing file name", iblock.Bnumber)
+	}
+	filename, err := ioutil.ReadFile(udevPath)
+	if err != nil {
+		return nil, fmt.Errorf("iscsi: GetIBlockUdev: Cannot read iblock from udev link :%s", udevPath)
+	}
+	iblock.Iblock = strings.TrimSpace(string(filename))
+
+	return &iblock, nil
+}
+
+// GetRBDMatch is getting the actual info to build up
+// the RBD data and match with the enable target
+func (rbd RBD) GetRBDMatch(sysDevicePath string, rbdNumber string, poolImage string) (r *RBD, err error) {
+
+	rbd.Name = "rbd_" + rbdNumber
+	rbd.Rnumber = rbdNumber
+
+	systemRbds, err := filepath.Glob(filepath.Join(sysDevicePath, "/devices/rbd/[0-9]*"))
+	if err != nil {
+		return nil, fmt.Errorf("iscsi: GetRBDMatch: Cannot find any rbd block")
+	}
+
+	for systemRbdNumber, systemRbdPath := range systemRbds {
+		var systemPool, systemImage string = "", ""
+		systemPoolPath := filepath.Join(systemRbdPath, "pool")
+		if _, err := os.Stat(systemPoolPath); os.IsNotExist(err) {
+			continue
+		}
+		bSystemPool, err := ioutil.ReadFile(systemPoolPath)
+		if err != nil {
+			continue
+		} else {
+			systemPool = strings.TrimSpace(string(bSystemPool))
+		}
+
+		systemImagePath := filepath.Join(systemRbdPath, "name")
+		if _, err := os.Stat(systemImagePath); os.IsNotExist(err) {
+			continue
+		}
+		bSystemImage, err := ioutil.ReadFile(systemImagePath)
+		if err != nil {
+			continue
+		} else {
+			systemImage = strings.TrimSpace(string(bSystemImage))
+		}
+
+		if matchRBD(fmt.Sprintf("%d", systemRbdNumber), rbd.Rnumber) &&
```

Comment:
I think `strconv.FormatInt(systemRbdNumber)` would be better here instead of `fmt.Sprintf`.

Reply:
Thanks, remove matchRBD all together and replace it with FormatInt(int64(systemRbdNumber) and inline string compare to reduce complexity.